### PR TITLE
add script-dump document

### PR DESCRIPTION
**Pull request by kukey**
Docs for https://github.com/valkey-io/valkey/pull/617

Change-Id: I9213aa113bcbf1337ac25e8f0540155e25eb1d05

### CODE REVIEW

File: topics/eval-intro.md [hunk 1]
```diff
@@ -222,6 +222,9 @@ These are:
 * `SCRIPT LOAD script`: this command registers the specified script in the Valkey script cache. 
   It is a useful command in all the contexts where we want to ensure that `EVALSHA` doesn't not fail (for instance, in a pipeline or when called from a [`MULTI`/`EXEC` transaction](transactions.md)), without the need to execute the script.
 
+* `SCRIPT DUMP sha1`: this command dump server-side script in the script cache.
```
Comment (madolson) [inline topics/eval-intro.md:225]:
```suggestion
* `SCRIPT SHOW`: this command shows the original source code for a script that is stored in the script cache. 
```

File: commands/script-dump.md [hunk 2]
```diff
@@ -0,0 +1,10 @@
+Returns the scripts in the script cache.
```
Comment (madolson) [inline commands/script-dump.md:1]:
```suggestion
Returns the original source code of a script in the script cache.
```

File: commands/script-dump.md [hunk 3]
```diff
@@ -0,0 +1,10 @@
...
+
+This command accepts one SHA1 digest and return a script if the script
+are already defined or not inside the script cache.
+This can be useful to debug. In some scenarios, the business may not be able to find the
+previously used Lua script and only have a SHA signature.
+Or there are multiple identical evalsha's args in monitor/slowlog,
+and admin is not able to distinguish the script body.
```
Comment (madolson) [inline commands/script-dump.md:8]:
```suggestion
This command accepts a SHA1 digest and returns the original script's source code if the script is present in the script cache.
This command is intended primarily for debugging, to introspect into the contents of a script when the user does not have access to script anymore.
One example is when an admin only has access to a script SHA from monitor or slowlog, and is unable to determine the contents of the script.
```
Code convention says one sentence per line. I also tried to simplify the wording a bit, feel free to take or leave some of my suggestions.

**Review by madolson — COMMENTED** [review]
Please update the file to `script-show`.

File: commands/script-dump.md [hunk 4]
```diff
@@ -0,0 +1,10 @@
+Returns the scripts in the script cache.
+
+This command accepts one SHA1 digest and return a script if the script
```
Comment (hwware) [inline commands/script-dump.md:3]:
```suggestion
This command accepts one SHA1 digest and returns a script if the script
```

File: commands/script-dump.md [hunk 5]
```diff
@@ -0,0 +1,10 @@
...
+are already defined or NOSCRIPT error if not inside the script cache.
```
Comment (hwware) [inline commands/script-dump.md:4]:
```suggestion
is already defined or NOSCRIPT error if not inside the script cache.
```

File: commands/script-dump.md (hunk 3 above)
Comment (zuiderkwast) [reply commands/script-dump.md]: English speakers: Is it "a SHA1" or "an SHA1"? Does it depend how you pronounce SHA1? [ess-aych-ay-1] vs [sha-1]

**Comment by zuiderkwast** [comment]
@kukey Can you update this PR to SCRIPT SHOW?

File: commands/script-dump.md (hunk 3 above)
Comment (kukey) [reply commands/script-dump.md]: @zuiderkwast SHA is Secure Hash Algorithm, so i think it is "a SHA1".

**Comment by kukey** [comment]
> @kukey Can you update this PR to SCRIPT SHOW?

OK

File: commands/script-dump.md (hunk 3 above)
Comment (kukey) [reply commands/script-dump.md]: @madolson @zuiderkwast thanks for your suggestions

File: commands/script-dump.md
Comment (zuiderkwast) [inline commands/script-dump.md:1]: The filename is wrong.

File: commands/script-show.md [hunk 6]
```diff
@@ -0,0 +1,7 @@
+Returns the original source code of a script in the script cache.
+
+This command accepts a SHA1 digest and returns the original script's source code if the script is present in the script cache.
+This command is intended primarily for debugging, to introspect into the contents of a script when the user does not have access to script anymore.
+One example is when an admin only has access to a script SHA1 from monitor or slowlog, and is unable to determine the contents of the script.
```
Comment (madolson) [inline commands/script-show.md:5]:
```suggestion
For example, an admin may only have access to a script's SHA1 from the monitor or slowlog and needs to determine the script's contents.
```

File: commands/script-show.md [hunk 7]
```diff
@@ -0,0 +1,7 @@
+Returns the original source code of a script in the script cache.
+
+This command accepts a SHA1 digest and returns the original script's source code if the script is present in the script cache.
+This command is intended primarily for debugging, to introspect into the contents of a script when the user does not have access to script anymore.
```
Comment (madolson) [inline commands/script-show.md:4]:
```suggestion
It is intended primary for debugging, allowing users to introspect the contents of a script when they no longer have direct access to it.
```

**Review by madolson — COMMENTED** [review]
Just some minor wording improvements I'll apply.

**Review by madolson — APPROVED** [review]

**Comment by zuiderkwast** [comment]
The title is still wrong and the committed message...
